community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/CypherCompiler.scala

```diff
@@ -83,9 +84,14 @@ object CypherCompilerFactory {
     val planner = CostBasedPipeBuilderFactory(monitors, metricsFactory, planningMonitor, clock, queryPlanner = queryPlanner,
       rewriterSequencer = rewriterSequencer, plannerName = pickedPlannerName, runtimeName = pickedRuntimeName)
     // falling back to legacy planner is allowed only when no cost-based planner is picked explicitly (e.g., COST, IDP)
-    val pipeBuilder = pickedPlannerName match {
-      case FallbackPlannerName =>
+    val pipeBuilder = (pickedPlannerName, useErrorsOverWarnings) match {
+      // fallback to RULE if specific planner was not specified
+      case (FallbackPlannerName, _) =>
         new LegacyVsNewExecutablePlanBuilder(new LegacyExecutablePlanBuilder(monitors, rewriterSequencer), planner, planBuilderMonitor)
+      // fallback to RULE if configuration does not require error if specified planner cannot be supported
+      case (_, false) =>
+        new LegacyVsNewExecutablePlanBuilder(new LegacyExecutablePlanBuilder(monitors, rewriterSequencer), planner, planBuilderMonitor)
```

Comment:
The behaviour (and code) of the suggested `WarningReportingExecutablePlanBuilder` is almost identical to the `LegacyVsNewExecutablePlanBuilder`, with the single addition of a warning, so we choose to simply pass a boolean into the latter to warn, or be silent.